src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/payment/MarkPaymentPaidPreparer.java

```diff
@@ -125,15 +119,58 @@ public PreSubmitCallbackResponse<AsylumCase> handle(
+    private void checkPaymentConditions(AppealType appealType, AsylumCase asylumCase, PreSubmitCallbackResponse<AsylumCase> callbackResponse, boolean isAipJourney, Optional<RemissionType> remissionType, Optional<RemissionType> lateRemissionType) {
+        Optional<RemissionDecision> remissionDecision = asylumCase.read(REMISSION_DECISION, RemissionDecision.class);
+
+        Optional<String> paPaymentType = asylumCase.read(PA_APPEAL_TYPE_PAYMENT_OPTION, String.class);
+        Optional<PaymentStatus> paymentStatus = asylumCase.read(PAYMENT_STATUS, PaymentStatus.class);
+        Optional<String> eaHuPaymentType = asylumCase.read(EA_HU_APPEAL_TYPE_PAYMENT_OPTION, String.class);
+        boolean isEaHuEu = List.of(EA, HU, EU).contains(appealType);
+        // old cases
+        if ((appealType == PA && remissionType.isEmpty() && paPaymentType.isEmpty())
+            || (isEaHuEu && remissionType.isEmpty()
+            && eaHuPaymentType.isEmpty())) {
+            callbackResponse.addError(NOT_AVAILABLE_LABEL);
+        }
+        if (isEaHuEu && (remissionType.isEmpty() || remissionType.get() == NO_REMISSION)
+            && eaHuPaymentType.isPresent() && eaHuPaymentType.get().equals("payNow")
+            && paymentStatus.isPresent() && paymentStatus.get() == PaymentStatus.PAID) {
+            callbackResponse.addError(NOT_AVAILABLE_LABEL);
+        }
+
+        if (appealType == PA && remissionType.isPresent()
+            && remissionType.get() == NO_REMISSION
+            && isAipJourney) {
+            paPaymentType
+                .filter(option -> option.equals("payLater"))
+                .ifPresent(s ->
+                    callbackResponse.addError(NOT_AVAILABLE_LABEL)
+                );
+        } else if ((remissionType.isPresent() && remissionType.get() != NO_REMISSION && !remissionDecision.isPresent())
+            || (lateRemissionType.isPresent() && !remissionDecision.isPresent())) {
+
+            callbackResponse.addError("You cannot mark this appeal as paid because the remission decision has not been recorded.");
+        } else if (isRemissionDecisionExistsAndApproved(remissionDecision)) {
+
+            callbackResponse.addError("You cannot mark this appeal as paid because a full remission has been approved.");
+        }
     }
 
     private boolean isRemissionDecisionExistsAndApproved(
         Optional<RemissionDecision> remissionDecision
     ) {
 
         return remissionDecision.isPresent()
-               && remissionDecision.get() == APPROVED;
+            && remissionDecision.get() == APPROVED;
+    }
+
+    private boolean isRemissionExistsAip(Optional<RemissionOption> remissionOption, Optional<HelpWithFeesOption> helpWithFeesOption) {
+        boolean isDlrmFeeRemission = featureToggler.getValue("dlrm-fee-remission-feature-flag", false);
+
+        return (remissionOption.isPresent() && remissionOption.get() != uk.gov.hmcts.reform.iacaseapi.domain.entities.RemissionOption.NO_REMISSION)
```

Comment:
it requires the full path like uk.gov.hmcts.reform.iacaseapi.domain.entities.RemissionOption.NO_REMISSION?

Reply:
Yes, it requires the whole path because we are using remissionOption.No_remission and RemissionType.No_Remission in the same file.